glue/plugins/wcs_autolinking/wcs_autolinking.py

```diff
@@ -60,43 +86,144 @@ def __init__(self, data1=None, data2=None, cids1=None, cids2=None):
+                slicing_axes_celestial1 = [cids1_celestial[0].axis, cids1_celestial[1].axis]
+                slicing_axes_celestial1 = sorted(slicing_axes_celestial1, key=str, reverse=False)
+
+                slicing_axes_celestial2 = []
+
+            elif not wcs1.has_celestial and wcs2.has_celestial:
+                slicing_axes_celestial1 = []
+
+                wcs2_celestial_axis_physical_types = wcs2.celestial.world_axis_physical_types
+                cids2 = data2.pixel_component_ids
+                cids2_celestial = [cids2[wcs2.wcs.naxis - wcs2.wcs.lng - 1],
+                                   cids2[wcs2.wcs.naxis - wcs2.wcs.lat - 1]]
+                slicing_axes_celestial2 = [cids2_celestial[0].axis, cids2_celestial[1].axis]
+                slicing_axes_celestial2 = sorted(slicing_axes_celestial2, key=str, reverse=False)
+
+            # Collect all apparently matching axes in two lists for the two wcs objects being linked up
+            if wcs1_celestial_axis_physical_types is not None:
+                wcs1_sliced_physical_types = wcs1_celestial_axis_physical_types
+            else:
+                wcs1_sliced_physical_types = []
+            slicing_axes1 = slicing_axes_celestial1
+            if wcs2_celestial_axis_physical_types is not None:
+                wcs2_sliced_physical_types = wcs2_celestial_axis_physical_types
+            else:
+                wcs2_sliced_physical_types = []
+            slicing_axes2 = slicing_axes_celestial2
+            for i, physical_type1 in enumerate(wcs1.world_axis_physical_types):
+                for j, physical_type2 in enumerate(wcs2.world_axis_physical_types):
+                    print('physical_type1', physical_type1)
+                    print('physical_type2', physical_type2)
+                    if physical_type1 == physical_type2:
+                        print('{0} matches with {1}'.format(physical_type1, physical_type2))
+                        if physical_type1 not in wcs1_sliced_physical_types:
+                            slicing_axes1.append(wcs1.world_n_dim - i - 1)
+                            wcs1_sliced_physical_types.append(physical_type1)
+                        if physical_type2 not in wcs2_sliced_physical_types:
+                            slicing_axes2.append(wcs2.world_n_dim - j - 1)
+                            wcs2_sliced_physical_types.append(physical_type2)
+
+            slicing_axes1 = sorted(slicing_axes1, key=str, reverse=True)
+            slicing_axes2 = sorted(slicing_axes2, key=str, reverse=True)
+
+            # Generate slices for the wcs slicing
+
+            slices1 = [slice(None)] * wcs1.world_n_dim
+            slices2 = [slice(None)] * wcs2.world_n_dim
+
+            slices1 = sorted(list(slices1), key=str)
+            slices2 = sorted(list(slices2), key=str)
+
+            for i in range(wcs1.world_n_dim):
+                if i in slicing_axes1:
+                    pass
+                else:
+                    slices1[i] = 0
```

Comment:
Why not simply 'if i not in slicing axes:' to avoid the empty first clause?

Reply:
Yup, agreed